bigbluebutton/bigbluebutton
# feat(plugin): asset persistence

### What does this PR do?

It does 2 things in general:

- Creates a new feature for the plugins to use: the asset-persistence;
- Fix the MaxFileSize error that was not happening;

The second point is there just so I can properly validate if the file size of the presentation that the person is sending is not bigger than expected.

### Motivation

This will allow us to send PDFs into BBB via plugin so that new ways of uploading an h5p file can exist - this is for the h5p-plugin.

### How to test
Just use the `sample-action-button-dropdown-plugin` in the plugin-sdk repository. It will upload automatically a sample presentation.

To test the expected error, one can decrease the number in `manifest.assetPersistence.maxFileSize` to say 100, which means it allows a file with a maximum of 100 Bytes (pretty much any file you send is greater than that).

### More

See demo:

https://github.com/user-attachments/assets/585a430c-89b2-42db-b894-0feebff28cdc

Closely related to SDK PR: https://github.com/bigbluebutton/bigbluebutton-html-plugin-sdk/pull/133

**There is only one feature left to implement here**
- [ ] validate maxUploadSizePerUser;

That is, the maximum file-size that a user can upload via the plugin, so that if we have 100MB for this config, the user can send 10 files of 10MB, for instance, or 20 files of 5MB, and so on.

This will be implemented once the issue https://github.com/bigbluebutton/bigbluebutton/issues/21776 is fixed, because this will introduce a new type of expected error in the presentation.
